test/unit/migration/migrator.test.ts

```diff
@@ -221,6 +221,60 @@ describe('MIGRATION: migrator', () => {
       expect(result1[0].methods.down.name).to.deep.equal('down');
     });
 
+    it('should include extension in the resolved object.', async () => {
+      const migrationPath = await mkdtemp();
+
+      const exampleJs = `
+      function up(db) {
+        return db.schema.createTable('demo_table', table => {
+          table.increments('id').primary()
+        });
+      }
+
+      function down(db) {
+        return db.schema.dropTable('demo_table');
+      }
+
+      module.exports = {
+        up: up,
+        down: down
+      }
+      `;
+
+      const exampleTs = `
+      export function up(db: any) {
+        return db.schema.createTable('test_demo', (table: any) => {
+          table.increments('id').primary()
+        });
+      }
+
+      export function down(db: any) {
+        return db.schema.dropTable('test_demo');
+      }
+      `;
+
+      // Populate migration files to a directory.
+      await Promise.all([
+        write(path.join(migrationPath, '0001_create_table_users.js'), exampleJs),
+        write(path.join(migrationPath, '0002_alter_table_users_add_gender.ts'), exampleTs),
+        write(path.join(migrationPath, '.gitignore'), '')
+      ]);
+
+      const result = await migratorService.resolveJavaScriptMigrations(migrationPath, 'js', true);
```

Comment:
How about we always include the extension for `.js` and `.ts` files and not make it configurable?

Reply:
@cham11ng @samirsilwal Thoughts?

Reply:
sounds good. easier shift from knex to sync-db if in use

Reply:
Totally agreed. I made it configurable to support backward compatibility but to allow migration filenames without extension doesn't make sense. I'll mark this as a bug in migrator and change accordingly.